benchmarks/README.md

```diff
@@ -0,0 +1,113 @@
+# Benchmarks 
+
+## Overview
+This is a basic framework for writing benchmarks to measure Delta's performance. It is currently designed to run benchmark on Spark running in an EMR cluster. However, it can be easily extended for other Spark-based benchmarks. To get started, first download/clone this repository in your local machine. Then you have to set up an EMR cluster and run the benchmark scipts in this directory. See the next section for more details.
+
+## Running benchmark
+
+### Prerequisites
+- An AWS account with necessary permissions to do the following:
+  - Manage EMR clusters for running the benchmark
+  - Read and write to an S3 bucket from the EMR cluster
+- A S3 bucket which will be used to generate the TPC-DS data in different storage formats.
+- A machine which has access to the AWS setup and where this repository has been downloaded or cloned. 
+
+### Create EMR cluster
+Create an EMR cluster that connects to the external Hive Metastore.  Here are the specifications of the EMR cluster required for running benchmarks.
+- Master - i3.2xlarge
+- Workers - i3.2xlarge (just 1 worker if you are just testing by running the 1GB benchmark).
+- No autoscaling, and default EBS storage.
+
+Once the EMR cluster is ready, note the following: 
+- Hostname of the EMR cluster master node.
+- PEM file for SSH into the master node.
+These will be needed to run the workloads in this framework. 
+
+### Test the cluster setup
+Navigate to your local copy of this repository and this benchmark directory. Then run the following steps.
+
+#### Run simple test workload
+Verify that you have the following information
+  - <HOST_NAME>: EMR cluster master node host name
+  - <PEM_FILE>: Local path to your PEM file for SSH into the master node.
+  - <BENCHMARK_PATH>: S3 path where tables will be created. Make sure your AWS credentials have read/write permission to that path.
+    
+Then run a simple table write-read test: Run the following in your shell. 
+    ```sh
+    ./run-benchmark.py --cluster-hostname <HOSTNAME> -i <PEM_FILE> --benchmark-path <BENCHMARK_PATH> --benchmark test 
```

Comment:
This is running with python 2 on my machine. I had to use `python3 run-benchmark.py --cluster-hostname ...`

Reply:
thats weird. .. i thought `#!/usr/bin/env python3` I have at the top of the run-benchmark.py should make it use python3.